**Notice (PR bugs + fixes)**
- **Critical (still open): `SECRET_KEY` generation can corrupt `.env`** at Makefile. If generated key contains `&` (or `|`), `sed` expands it and produces broken values like repeated `SECRET_KEY=...`.  
  **Fix can be applied:** escape replacement before `sed` (`ESCAPED_KEY=$(printf '%s' "$KEY" | sed 's/[&|]/\\&/g')`) or replace this step with a tiny Python rewrite script.
- **Previously failing on paths with spaces** at Makefile.  
  **Fix can be applied* venv executables are quoted.
- **Previously failing when host Postgres occupied 5432** due to hardcoded Django DB port.  
  **Fix can be applied** Django now reads `DB_PORT` in settings.py.
- **Previously inconsistent infra env loading** in local dev targets.  
  **Fix can be applied** `dev-setup/dev/dev-stop` now pass `--env-file backend/.env` in Makefile.
- **Docs mismatch:** local-dev port in DEPLOYMENT.md assumes `5432`; users with local Postgres may need `DB_PORT=55432`. Add a troubleshooting note.
- **Non-blocking warning on Apple Silicon:** default PostGIS image is amd64; advise setting ARM image via .env.example.

Error 1: (make dev-setup)
[1/7] Python virtual environment...
[2/7] Installing backend dependencies...
make: /Users/zeynep/Development/SWE/SWE: No such file or directory
make: *** [dev-setup] Error 1

Error 2: (db connection)
 ! db The requested image's platform (linux/amd64) does not match the detected host platform (linux/arm64/v8) and no specific platform was requested 0.0s 
  Waiting for database to accept connections (15s)...
[5/7] Running Django migrations...
Traceback (most recent call last):
  File "**/SWE-574-3/backend/.venv/lib/python3.12/site-packages/django/db/backends/base/base.py", line 279, in ensure_connection
    self.connect()

--- a/DEPLOYMENT.md
+++ b/DEPLOYMENT.md
@@ -33,6 +33,20 @@ make dev-stop
 
 **Demo accounts** (password: `demo123`): `elif@demo.com`, `cem@demo.com`, `moderator@demo.com` (admin)
 
+> **Troubleshooting — port conflicts**
+>
+> If you already have PostgreSQL or Redis running natively on your machine, the default ports (5432 / 6379) will be taken. Override them in `backend/.env`:
+> ```dotenv
+> DB_PORT=55432    # Docker will expose PostGIS on this host port
+> REDIS_PORT=56379
+> ```
+> `make dev-setup` / `make dev` pass `--env-file backend/.env` to Docker Compose, so the containers will bind to the ports you specify.
+>
+> **Apple Silicon (ARM64):** The default PostGIS image is AMD64 and will show a platform warning. Set in `backend/.env`:
+> ```dotenv
+> POSTGIS_IMAGE=imresamu/postgis-arm64:alpine-ver20251223-6b15838-2026w09
+> ```
+
 | URL | Service |
 |-----|---------|
 | http://localhost:5173 | Frontend |

--- a/Makefile
+++ b/Makefile
@@ -4,8 +4,9 @@
 # ── Local dev config ──────────────────────────────────────────────────────────
 PYTHON  ?= python3
 VENV     = backend/.venv
-PIP      = $(CURDIR)/$(VENV)/bin/pip
-PYEXEC   = $(CURDIR)/$(VENV)/bin/python
+# Quoted so paths containing spaces (e.g. "My Projects/SWE-574-3") don't break
+PIP      = "$(CURDIR)/$(VENV)/bin/pip"
+PYEXEC   = "$(CURDIR)/$(VENV)/bin/python"
 
 help: ## Show this help message
 	@echo 'Usage: make [target]'
@@ -26,14 +27,19 @@ dev-setup: ## One-time local setup: venv, deps, infra, migrate, demo data
 	@if [ ! -f backend/.env ]; then \
 		cp .env.example backend/.env; \
 		sed -i '' 's|DB_HOST=localhost|DB_HOST=127.0.0.1|g' backend/.env; \
-		KEY=$$(cd backend && $(PYEXEC) -c "from django.core.management.utils import get_random_secret_key; print(get_random_secret_key())"); \
-		sed -i '' "s|SECRET_KEY=.*|SECRET_KEY=$$KEY|" backend/.env; \
+		$(PYEXEC) -c " \
+import pathlib, sys; \
+from django.core.management.utils import get_random_secret_key; \
+p = pathlib.Path('backend/.env'); \
+lines = ['SECRET_KEY=' + get_random_secret_key() if l.startswith('SECRET_KEY=') else l for l in p.read_text().splitlines()]; \
+p.write_text('\n'.join(lines) + '\n') \
+"; \
 		echo "  Created backend/.env"; \
 	else \
 		echo "  backend/.env already exists, skipping."; \
 	fi
 	@echo "\033[1;34m[4/7] Starting infra (PostGIS + Redis)...\033[0m"
-	@docker compose -f docker-compose.infra.yml up -d
+	@docker compose -f docker-compose.infra.yml --env-file backend/.env up -d
 	@echo "  Waiting for database to accept connections (15s)..."
 	@sleep 15
 	@echo "\033[1;34m[5/7] Running Django migrations...\033[0m"
@@ -48,7 +54,7 @@ dev-setup: ## One-time local setup: venv, deps, infra, migrate, demo data
 
 dev: ## Start local dev: infra + backend (8000) + frontend (5173) in parallel
 	@echo "\033[1;34m→ Starting infra...\033[0m"
-	@docker compose -f docker-compose.infra.yml up -d
+	@docker compose -f docker-compose.infra.yml --env-file backend/.env up -d
 	@echo "\033[1;34m→ Starting backend (http://localhost:8000) and frontend (http://localhost:5173)...\033[0m"
 	@echo "  Press Ctrl+C to stop both.\n"
 	@trap 'kill 0; exit 0' INT TERM; \
@@ -58,7 +64,7 @@ dev: ## Start local dev: infra + backend (8000) + frontend (5173) in parallel
 
 dev-stop: ## Stop local infra (PostGIS + Redis containers)
 	@echo "\033[1;34m→ Stopping infra...\033[0m"
-	@docker compose -f docker-compose.infra.yml down
+	@docker compose -f docker-compose.infra.yml --env-file backend/.env down
 	@echo "\033[1;32m✓ Infra stopped.\033[0m"
 
 # ─────────────────────────────────────────────────────────────────────────────

--- a/backend/hive_project/settings.py
+++ b/backend/hive_project/settings.py
@@ -149,7 +149,7 @@ DATABASES = {
         'USER': os.environ.get('DB_USER'),
         'PASSWORD': os.environ.get('DB_PASSWORD'),
         'HOST': os.environ.get('DB_HOST'),
-        'PORT': '5432',
+        'PORT': os.environ.get('DB_PORT', '5432'),
     }
 }
 

